pmm-app/src/react-plugins-deps/components/Table/Table.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { ReactElement } from 'react';
+import { useRowSelect, useTable } from 'react-table';
+import { Spinner, useTheme } from '@grafana/ui';
+import { getStyles } from './Table.styles';
+
+interface TableProps {
+  columns: object[];
+  data: object[];
+  noData?: ReactElement;
+  actionPanel?: (selected: any[]) => ReactElement;
+  loading?: boolean;
+  rowKey?: (rec: any) => any;
+}
+
+const TableCheckbox = props => {
+  return (
+    <label className="checkbox-container checkbox-container--main no-gap">
+      <input type="checkbox" {...props} />
+      <span className="checkbox-container__checkmark"></span>
+    </label>
+  );
+};
+
+function Table({ columns, data, actionPanel, noData, loading, rowKey }: TableProps) {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+  // @ts-ignore
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow, selectedFlatRows } = useTable(
+    {
+      columns,
+      data,
+    },
+    useRowSelect,
+    hooks => {
+      if (actionPanel) {
+        // @ts-ignore
+        hooks.visibleColumns.push(columns => [
+          {
+            id: 'selection',
+            Header: props => (
+              <div data-qa="select-all">
+                {/*
+                  // @ts-ignore */}
```

Comment:
Pls don't disable typescript

Reply:
I guess that react-table have some issues with interfaces when you add additional functionality hooks, such as select rows functionality